### PR TITLE
:tada: Denops v7

### PR DESCRIPTION
**Pull request by lambdalisue**
- Deno: 1.38.x -> ~~1.43.x~~ 1.45.x
- Vim: 9.0.2189 -> ~~9.1.0399~~ 9.1.0448
- Neovim: 0.9.4 -> ~~0.9.5~~ 0.10.0

## Ref

- https://github.com/vim-denops/denops.vim/pull/344
- https://github.com/vim-denops/deno-denops-core/pull/8
- https://github.com/vim-denops/deno-denops-std/pull/246
- https://github.com/vim-denops/deno-denops-test/pull/25

## Summary by CodeRabbit

- **New Features**
  - Introduced asynchronous request interruption with `denops#interrupt(...)`.
  - Added plugin unload functionality with improved error handling.
  - Implemented new `wait` function for polling asynchronous operations.
  - Added support for shared server management in testing environments.
  - Introduced dummy plugins for testing purposes.

- **Enhancements**
  - Updated version support for Deno, Vim, and Neovim across documentation and plugin management.
  - Improved error handling for job executions.
  - Enhanced job handling logic for Vim and Neovim RPC modules.
  - Refined connection management and reconnect logic for the server.
  - Introduced mock utility functions for testing purposes.

- **Tests**
  - Introduced new test cases for various Denops plugins and scenarios.
  - Enhanced the testing framework with new assertions and improved coverage for the `Service` class.

- **Documentation**
  - Updated Denops usage settings and variable descriptions in documentation, including a new recommended section.

### CODE REVIEW

File: deno.jsonc [hunk 1]
```diff
@@ -1,11 +1,10 @@
...
-  "lock": false,
   "tasks": {
+    "check": "deno check **/*.ts",
     "test": "deno test -A --parallel --shuffle --doc",
     "test:coverage": "deno task test --coverage=.coverage",
-    "check": "deno check ./**/*.ts",
     "coverage": "deno coverage .coverage --exclude=cli.ts --exclude=worker.ts --exclude=testdata/",
-    "upgrade": "deno run -q -A https://deno.land/x/molt@0.14.2/cli.ts ./**/*.ts",
-    "upgrade:commit": "deno task -q upgrade --commit --prefix :package: --pre-commit=fmt"
+    "update": "deno run --allow-env --allow-read --allow-write=. --allow-run=git,deno --allow-net=jsr.io,registry.npmjs.org jsr:@molt/cli ./*.ts",
```
Comment (Milly) [inline deno.jsonc:7]:
```suggestion
    "update": "deno run --allow-env --allow-read --allow-write=. --allow-run=git,deno --allow-net=jsr.io,registry.npmjs.org jsr:@molt/cli **/*.ts",
```

**Review by Milly — CHANGES_REQUESTED** [review]

**Comment by lambdalisue** [comment]
https://uki00a.github.io/deno-weekly/articles/2024/06/16.html#node-api

Node API is fixed so we can remove https://github.com/vim-denops/denops.vim/pull/344/commits/fc956a92d3c558dbf926d736133da013c9beec4e

File: denops/@denops-private/service.ts [hunk 2]
```diff
@@ -125,13 +146,41 @@ export class Service implements Disposable {
...
 }
 
+type PluginModule = {
+  main: Entrypoint;
+};
+
 class Plugin {
   #denops: Denops;
+  #loadedWaiter: Promise<void>;
+  #disposable: AsyncDisposable = voidAsyncDisposable;
```
[the remark this answers is not in the document]
Comment (lambdalisue) [reply denops/@denops-private/service.ts:183]: No. The value is initialized in the constructor and that is a common pattern I think.

**Review by Milly — DISMISSED** [review]

**Comment by lambdalisue** [comment]
https://github.com/denoland/deno/releases/tag/v1.45.0

**Review by Milly — APPROVED** [review]

**Pull request merged**